src/app_service/service/message/service.nim

```diff
@@ -577,6 +584,34 @@ QtObject:
     except Exception as e:
       error "error: ", procName="pinUnpinMessage", errName = e.name, errDesription = e.msg
 
+  proc markMessageAsUnread*(self: Service, chatId: string, messageId: string) = 
```

Comment:
This procedure is calling a status-go endpoint that accesses the database.
In such cases we usually implement an "async task", you can find examples in neighbour file: `src/app_service/service/message/async_tasks.nim`.

Though I think we fixed lots of issues with databases, maybe it's not the case anymore?
cc @osmaczko 

_____

Also, while this procedure is syncronous, it doesn't make much sense to emit a `SIGNAL_MESSAGE_MARKED_AS_UNREAD` signal with result. We could just return it from the proc. But we would need to keep the signal for an async task ofc.

Reply:
That's fair, let me update the code to have an async call.

Reply:
That's a good point, @igor-sirotin. Ideally, everything that interacts with the database should be asynchronous; nothing has changed on that topic.